packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -1218,6 +1227,52 @@ function updateTransition(
   return [startTransition, isPending];
 }
 
+let clientId: number = 0;
+
+function mountUniqueId(): string | IdObject {
+  const [id, setId] = mountState(() => {
+    if (getIsHydrating()) {
+      return {
+        $$typeof: REACT_OPAQUE_OBJECT_TYPE,
+        _setId: () => {
+          setId(() => 'c_' + (clientId++).toString());
+        },
+        toString() {
+          throw Error(
+            'The object passed back from useUniqueID is meant to be passed through to ' +
+              'attributes only. Do not directly modify the output.',
+          );
+        },
+      };
+    } else {
+      return 'c_' + (clientId++).toString();
+    }
+  });
+  return id;
+}
+
+function updateUniqueId(): string | IdObject {
+  const [id, setId] = updateState(() => {
+    if (getIsHydrating()) {
+      return {
+        $$typeof: REACT_OPAQUE_OBJECT_TYPE,
+        _setId: () => {
+          setId(() => 'c_' + (clientId++).toString());
+        },
+        toString() {
+          throw Error(
+            'The object passed back from useUniqueID is meant to be passed through to ' +
+              'attributes only. Do not directly modify the output.',
+          );
+        },
+      };
+    } else {
+      return 'c_' + (clientId++).toString();
+    }
```

Comment:
The body of this function looks identical to the mount function. Should we try to share it across mount and update?

Reply:
The update shouldn’t need to create the initializer (normally). Can we short cut this and avoid creating the unnecessary closure and array?